### PR TITLE
Update setting docs

### PR DESCRIPTION
**Pull request by PhilippeCollin**
Update settings documentation

### CODE REVIEW

**Review by mbarak — APPROVED** [review]

File: schemas/theme/setting.json [hunk 1]
```diff
@@ -258,7 +268,7 @@
           "markdownDescription": "A setting of type `collection_list` outputs a collection picker field that's automatically populated with the available collections for the store. You can use these fields to capture multiple collections, such as a group of collections to feature on the homepage.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/settings/input-settings#collection_list)"
         },
         "limit": {
-          "type": "number",
+          "type": "integer",
```
Comment (PhilippeCollin) [inline schemas/theme/setting.json:271]: tiny drive-by improvement, doesn't make sense for limits to be floating point

**Comment by kevinruscoe** [comment]
@PhilippeCollin Quick question RE this, you added the "metaobject_type" type and the docs for it, but this isn't usable setting within a liquid schema?

If I add

```
{
          "label": "Logo",
          "id": "logo",
          "type": "metaobject",
          "metaobject_type": "image"
}
```

I get an error `setting with id="logo" type is invalid`
